.golangci.yml

```diff
@@ -18,54 +18,46 @@
 linters-settings:
   lll:
     line-length: 170
+  goconst:
+    ignore-tests: true
 linters:
-  disable-all: true
-  enable:
-    - asciicheck
-    - bodyclose
-    - deadcode
-    - depguard
-    - dogsled
-    - durationcheck
-    - errcheck
-    - errname
-    - errorlint
-    - exportloopref
-    - forcetypeassert
-    - gocritic
-    - gofmt
+  enable-all: true
```

Comment:
I changed in a previous commit here from enable-all (denylisting) to disable-all (allowlisting), because otherwise every time we try to upgrade golangci-lint it will introduce a new set of linters which weren't checked before and thus upgrading golangci-lint will become painful. I thought using `disable-all` and allowlisting makes upgrading the golangci-lint action easier.

So, do you think it's not a good idea and we should catch up with new additions to linters along with upgrade of the action?

Reply:
I can revert to the `disable-all` mode, I don't have any strong opinion here.

The reason I did it the other way is to be sure we are not missing any important lint (as example. the `gosec` one was missing) so I did try to run it with all the possible lints and then disabling those that are just noise.

@astefanutti  @squakez @tadayosi since I'm not contributing to the project very often now, I let you decide

Reply:
On one hand with `disable-all`, it's less work to upgrade, but we would lag behind what would be the "default" / de-facto standard set of linters, and we would create a "debt" over time that would have to be pay by some larger one-time efforts, on the other hand with `enable-all`, there is some extra work possible each time we upgrade, which may biased us from not upgrading, but we would inherit from the "default", possibly better set of linters.

I'd be inclined toward trusting `golangci-lint` and trying with `enable-all`, so we dilute the possible overhead, keeping in mind that we can always revert to `disable-all` if that does not work.

Reply:
can we merge this PR then ?

Reply:
I think so, was waiting for your final confirmation :)

Reply:
Fair enough. Let's use `enable-all` and see how it works.